src/Video/ParticleSystemRenderer.hpp

```diff
@@ -0,0 +1,148 @@
+#pragma once
+#include <glm/glm.hpp>
+#include <Video/Shader/Shader.hpp>
+#include <Video/Shader/ShaderProgram.hpp>
+
+#include "linking.hpp"
+
+namespace Particles {
+    struct ParticlePos {
+        float x;
+        float y;
+        float z;
+        float w;
+    };
+
+    struct ParticleVelocity {
+        float vx;
+        float vy;
+        float vz;
+        float life;
+    };
+
+    struct ParticleColor {
+        float cx;
+        float cy;
+        float cz;
+        float ca;
+    };
+}
+
+
+struct ParticleModelMatrix {
+    glm::mat4 m = glm::mat4(1);
+};
+
+namespace Video {
+    class Texture2D;
+
+    class ParticleSystemRenderer {
+        public:
+        struct EmitterSettings {
+            /// Number of particles.
+            int nr_particles = 1024;
+
+            /// Position.
+            glm::vec3 worldPos;
+
+            /// Size.
+            glm::vec2 size;
+
+            /// Life (in seconds).
+            float life;
+
+            /// Lifetime (in seconds).
+            float lifetime = 10;
+
+            /// Initial velocity.
+            glm::vec3 velocity;
+
+            /// Start, mid and end of life alpha of particle.
+            glm::vec3 alpha;
+
+            /// Color of the particle.
+            glm::vec3 color = glm::vec3(0, 0, 0);
+
+            /// Fire rate.
+            float rate = 1.0f;
+
+            /// Texture index (for the texture atlas, left to right, top to bottom indexing)
+            int textureIndex = 0;
+
+            /// Multiply velocity.
+            float velocityMultiplier = 10.0f;
+
+            /// Gives spread to the particles
+            int spread = 1;
+
+            /// Used to give random velocity to particles at all time.
+            glm::vec3 randomVec = glm::vec3(1, 1, 1);
+
+            /// Scale.
+            float scale = 1.0f;
+
+            /// Particle mass.
+            float mass = 0.01f;
+
+            /// Alpha control
+            float alpha_control = 1.0f;
+
+            /// Nr Of new particles per emitt, max is 32.
+            int nr_new_particles = 31;
+        };
+
+        EmitterSettings emitterSettings;
+
+        VIDEO_API ParticleSystemRenderer();
+        VIDEO_API ParticleSystemRenderer(int count);
+        VIDEO_API ~ParticleSystemRenderer();
+
+        VIDEO_API void Init();
+
+        /// Creates storage buffers for particles.
+        VIDEO_API void CreateStorageBuffers();
+
+        /// Particles are sent to the compute shader and we compute there the new positions/velocities.
```

Comment:
Grammar mistake: `there` -> `their`. The next line seems to be residue from something else. Misaligned asterisks.